File: db/event_attachment.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_attachment.
*/

class Event_Attachment extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "event_attachment";
    $this->order = "";
    $this->domain = "event";
    $this->field['event_attachment_id']['type'] = 'SERIAL';
    $this->field['event_attachment_id']['not_null'] = true;
    $this->field['attachment_type_id']['type'] = 'INTEGER';
    $this->field['attachment_type_id']['not_null'] = true;
    $this->field['event_id']['type'] = 'INTEGER';
    $this->field['event_id']['not_null'] = true;
    $this->field['mime_type_id']['type'] = 'INTEGER';
    $this->field['mime_type_id']['not_null'] = true;
    $this->field['filename']['type'] = 'VARCHAR';
    $this->field['filename']['length'] = 256;
    $this->field['title']['type'] = 'VARCHAR';
    $this->field['title']['length'] = 64;
    $this->field['data']['type'] = 'BYTEA';
    $this->field['data']['not_null'] = true;
    $this->field['f_public']['type'] = 'BOOL';
    $this->field['f_public']['not_null'] = true;
    $this->field['f_public']['default'] = true;
    $this->field['event_attachment_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/event_attachment.rb
module Momomoto
  class Event_attachment < Base
    def initialize
      super
      @domain = 'event'
      @log_changes = true
      @fields = {
        :event_attachment_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :attachment_type_id => Datatype::Integer.new( {:not_null=>true} ),
        :event_id => Datatype::Integer.new( {:not_null=>true} ),
        :mime_type_id => Datatype::Integer.new( {:not_null=>true} ),
        :filename => Datatype::Varchar.new( {:length=>256} ),
        :data => Datatype::Bytea.new( {:not_null=>true} ),
        :f_public => Datatype::Bool.new( {:not_null=>true, :default=>true} ),
        :title => Datatype::Varchar.new( {:length=>256} ),
        :pages => Datatype::Integer.new( {} ),
        :last_modified => Datatype::Timestamp.new( {:with_timezone=>true, :not_null=>true, :default=>true, :auto_update=>true} ),
        :last_modified_by => Datatype::Integer.new( {} )
      }
    end
  end
end
